docs/DesignDocs/oesign_digest_signing_support.md

```diff
@@ -0,0 +1,125 @@
+# Support for Digest Signing in oesign
+
+This document proposes updates to the command line interface of the public oesign tool to support 
+a digest signing workflow for production enclaves. 
+
+
+## Motivation
+
+Per issue #2089:
+
+> oesign takes the signing key and enclave to sign in a single step today, which is not amenable for production signing scenarios as the owner of the private key is usually in a locked-down environment that is separate from the build pipeline.
+>
+> Similar to Intel's sgx_sign tool, oesign needs to support _digest signing_, where:
+>
+> 1. The lower trust build environment invokes oesign to generate a digest of the enclave
+> 2. Only the digest is sent to the high trust production signing environment for signing
+> 3. The resulting signature is returned to the build environment where oesign is invoked again to embed the signature to the enclave binary
+
+## User Experience
+
+To sign an OE enclave binary _my_enclave_ with the optional metadata specified in _my_enclave.conf_ and the _enclave_signing_key.pem_ owned by a different signing authority:
```

Comment:
Maybe adding some diagrams here would help clarify the flow.

Reply:
Added.